clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -977,18 +1058,20 @@ void handleCoordinatorReady() {
                 initProducerIdVersion.maxVersion() >= 3;
     }
 
-    private void transitionTo(State target) {
```

Comment:
Was it not possible to have a helper with just the target and CallingThread? Seeing all the nulls as parameters was a little confusing.

Reply:
I overloaded `transitionTo` so as to let callers with `null` use the more concise version.